File: LeetCode/Python/Easy/293-Flip-Game/solution.py
class Solution:
    def generatePossibleNextMoves(self, currentState: str) -> List[str]:
        result=list()

        n=len(currentState)

        can_flip=False

        for index in range(1, n):
            if currentState[index]=='+' and currentState[index - 1]=='+':
                can_flip=True

                break
        
        if not can_flip:
            return result

        indices_to_flip=list()

        for index in range(1, n):
            if currentState[index]=='+' and currentState[index - 1]=='+':
                indices_to_flip.append((index - 1, index))

        for count in range(0, len(indices_to_flip)):
            signs=[currentState[index] for index in range(0, n)]

            first_index=indices_to_flip[count][0]

            second_index=indices_to_flip[count][1]

            signs[first_index]='-'

            signs[second_index]='-'

            next_state=''.join(signs)

            result.append(next_state)

        return result

File: LeetCode/Rust/Easy/293-Flip-Game/solution.rs
impl Solution
{
    pub fn generate_possible_next_moves(current_state: String) -> Vec<String>
    {
        let mut result: Vec<String>=Vec::new();

        let n: usize=current_state.len();

        let mut can_flip: bool=false;

        let characters: Vec<char>=current_state.chars().collect::<Vec<char>>();

        for index in 1..n
        {
            if characters[index]=='+' && characters[index - 1]=='+'
            {
                can_flip=true;

                break;
            }
        }

        if !can_flip
        {
            return result;
        }

        let mut indices_to_flip: Vec<(usize, usize)>=vec![];

        for index in 1..n
        {
            if characters[index]=='+' && characters[index - 1]=='+'
            {
                indices_to_flip.push((index - 1, index));
            }
        }

        for count in 0..indices_to_flip.len()
        {
            let mut signs: Vec<char>=characters.clone();

            let first_index: usize=indices_to_flip[count].0;

            let second_index: usize=indices_to_flip[count].1;

            signs[first_index]='-';

            signs[second_index]='-';

            let next_state: String=signs.into_iter().collect::<String>();

            result.push(next_state);
        }

        result
    }
}